experimental/ModelBuilder/test/CMakeLists.txt

```diff
@@ -0,0 +1,276 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+iree_add_all_subdirs()
+
+file(GLOB _GLOB_X_CPP LIST_DIRECTORIES false RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} CONFIGURE_DEPENDS *.cpp)
+file(GLOB _GLOB_BENCHX LIST_DIRECTORIES false RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} CONFIGURE_DEPENDS Bench*)
+list(REMOVE_ITEM _GLOB_X_CPP ${_GLOB_BENCHX})
+iree_lit_test_suite(
+  NAME
+    lit
+  SRCS
+    "${_GLOB_X_CPP}"
+  DATA
+    experimental::ModelBuilder::test::test-dot-prod
+    experimental::ModelBuilder::test::test-matmul-vulkan
+    experimental::ModelBuilder::test::test-mnist-jit
+    experimental::ModelBuilder::test::test-simple-jit
+    experimental::ModelBuilder::test::test-simple-jit-vulkan
+    experimental::ModelBuilder::test::test-simple-mlir
+    experimental::ModelBuilder::test::test-vec-to-gpu
+#   experimental::ModelBuilder::test::test-vector-transfers-jit
```

Comment:
?

Reply:
I commented out this project as a workaround (I have a comment for it below). Shared library don't seem to work with bazel_to_cmake and this project relies on being able to create a shared library. This is something I can investigate after unless you have a solution simple enough that it could be added to this patch?

Reply:
I added limited support for shared libraries in https://github.com/google/iree/pull/2197:

https://github.com/google/iree/blob/6b7b71bdcf024caf8b302b06a9ee608f9126a997/iree/base/BUILD#L161-L166

https://github.com/google/iree/blob/6b7b71bdcf024caf8b302b06a9ee608f9126a997/iree/base/CMakeLists.txt#L182-L200

There is no `bazel_to_cmake` handling yet.

Reply:
Ah yeah I just meant it might be worth a comment here, even just "disabled, see below". Was reading quickly and wondering why there was a commented-out thing without explanation.

Reply:
I added a comment. 

@ScottTodd I tried to adding SHARED to the library but I'm still getting "undefined reference to `main'" error. I need to investigate more.

Reply:
Note that the Bazel `cc_binary` corresponds to a CMake `iree_cc_library` (unlike with regular binaries). This PR currently has `iree_cc_binary` commented out.

Reply:
That was it. Totally missed that part. Thanks @ScottTodd ! 
I updated the PR and re-enabled the both library I had disabled.